espwrap/adaptors/sendgrid_v3.py

```diff
@@ -61,109 +61,108 @@ def add_tags(self, *tags):
-                message.bcc = list_bcc
-
-            #Content (Text)
-            if self.body[MIMETYPE_TEXT]:
-                content = Content(
-                    MimeType.text,
-                    self.body[MIMETYPE_TEXT]
-                )
-                message.content = content
-
-            #Content (Html)
-            if self.body[MIMETYPE_HTML]:
-                content = Content(
-                    MimeType.html,
-                    self.body[MIMETYPE_HTML]
-                )
-                message.content = content
-
-            #Attachment
-            if self.attachments:
-                for file_name, file_path_or_string in self.attachments.iteritems():
-                    attachment = Attachment()
-                    encoded = base64.b64encode(str(file_path_or_string))
-                    attachment.file_content = FileContent(encoded)
-                    attachment.file_name = FileName(file_name)
-                message.attachment = attachment
-
-            #Webhook Data
-            if self.webhook_data:
-                for key, val in self.webhook_data.items():
-                    message.custom_arg = CustomArg(key, val)
-
-            #Metadata
-            if hasattr(self, 'metadata'):
-                for key, val in self.metadata.items():
-                    message.custom_arg = CustomArg(key, val)                       
-
-            #Template Name
-            if self.template_name:
-                message.custom_arg = CustomArg('template_name', self.template_name)
-
-            #Tracking
-            tracking_settings = TrackingSettings()            
-            #Opens
-            if self.track_opens:
-                tracking_settings.open_tracking = OpenTracking(
-                    self.track_opens,
-                    OpenTrackingSubstitutionTag("open_tracking")
-                )
-            #Clicks
-            if self.track_clicks:
-                tracking_settings.click_tracking = ClickTracking(
-                    self.track_clicks,
-                    self.track_clicks
-                )
-            message.tracking_settings = tracking_settings
-
-            #Set personalization fields
-            for p, recip in enumerate(subgrp):
-                message.from_email = From(self.from_addr)
-                message.to = To(recip['email'], recip['name'], p=p)
-                message.subject = Subject(self.subject)
-
-                #Global Subs
-                for key, val in self.global_merge_vars.items():
-                    new_key = '{1}{0}{2}'.format(key, *self.delimiters)
-                    message.substitution =  Substitution(new_key, val, p=p)
-
-                #Substitutions
-                for key, val in recip['merge_vars'].items():
-                    new_key = '{1}{0}{2}'.format(key, *self.delimiters)
-                    message.substitution = Substitution(new_key, val, p=p)
+            substitutions = subgrp[0]['merge_vars']
```

Comment:
Are we sure subgrp will always have at least one item here?

Reply:
I'm not really sure if subgrp is always at least one. I'm bit unfamiliar with email sending however I made changes to avoid this potential issue.